### ISSUE TITLE
New release for haxe 3.2?

### ISSUE THREAD

**Issue by andyli** [HaxeFoundation/neko#69]
As title. IIRC there are some haxe features introduced between 3.1.3 and 3.2.0 require a new version of neko.

**Comment by ConstNW** [comment]
does #31 complete already?

**Comment by ncannasse** [comment]
The unicode part is sadly not complete/tested, but bytes thing is ok. However it requires compiling with -D neko_v21 (for neko 2.1) to activate

**Comment by andyli** [comment]
Can we postpone the unicode part to neko 2.2 and get 2.1 out first for Haxe 3.2.0?

**Comment by andyli** [comment]
No?
Without an official release it would be hard to update the homebrew/chocolatey/ubuntu ppa packages.

**Comment by ncannasse** [comment]
I wish I had time to make a release :) I'm very very busy atm, maybe @Simn can help?

**Comment by Simn** [comment]
I wouldn't even know where to start.

**Comment by andyli** [comment]
@ncannasse I hope you have time to do a release soon. There are critical bugs (e.g. #39) fixed but the fix has never released.

I will go ahead and try to update the debian neko package. If there no serious problem, we can make a neko 2.1 release and I can submit it to debian.
A new release should also help @ppenzin's work on #84, in the sense that he can package the version that contains all his recently merged commits.

**Comment by thomasuster** [comment]
I would love to try neko out again now that the code limit was raised, so if anyone could help out @ncannasse I'd be grateful.

**Comment by ppenzin** [comment]
It is always better to use a release than just a commit (even though that might work too), so it would probably help me submit the port.

Let me know if I can help with speeding things up.

**Comment by ncannasse** [comment]
I did all the necessary steps last week end, now I only have to build a release. I still would like to package a mod_neko24 for Apache 2.4 since it's not compatible with Apache 2.2

**Comment by ppenzin** [comment]
Ping.

Anything I can do to help? Does anyone have an estimate when will the 
new release go out?

**Comment by ibilon** [comment]
In #112 @Simn said a new version would be released along haxe 3.3,
any estimation for when that'll be?

Because #86 isn't released some people on linux are loosing haxelib after a reboot or when cron launches a preload/strip http://community.openfl.org/t/roblem-with-install-haxe-sh-on-linux-mint-17-3-rosa/7110/

**Comment by Simn** [comment]
We need a neko release before we can make the haxe-3.3.0-rc1 release.

**Comment by ncannasse** [comment]
@pperidont are you all green with your SSL contributions to neko? I wouldn't want to have to update the release :)

**Comment by pperidont** [comment]
There's one last thing I would like to fix before a final release, it's thread safety on all platforms.
It will be useful for multithreads servers using SSL (Tora for example).

On Mac/Linux, it's simple, mbedTLS provides an implementation using pthread. It only requires some modifications in mbedTLS config.h. You can use this script to download, compile and install the lib before compiling neko :

```bash
wget https://tls.mbed.org/download/mbedtls-2.2.1-apache.tgz
tar xzf mbedtls-2.2.1-apache.tgz
cd mbedtls-2.2.1
sed -i "s/\/\/#define MBEDTLS_THREADING_PTHREAD/#define MBEDTLS_THREADING_PTHREAD/; s/\/\/#define MBEDTLS_THREADING_C/#define MBEDTLS_THREADING_C/; s/#define MBEDTLS_SSL_PROTO_SSL3/\/\/#define MBEDTLS_SSL_PROTO_SSL3/" include/mbedtls/config.h
SHARED=1 make lib
sudo make install
```

On Windows, we need a mbedTLS.lib with thread safety implemented. I will try to do that.

**Comment by andyli** [comment]
@pperidont Do you know whether the [mbedtls Debian package](https://packages.debian.org/source/sid/mbedtls) will just work? It's also available on [Ubuntu 16.04](http://releases.ubuntu.com/16.04/), which is currently in beta 2.

**Comment by pperidont** [comment]
@andyli The Debian package seems OK according to 01_config.patch. I think we can use it with dynamic linking for Debian/Ubuntu packages when it's available.

On Mac, the brew package did not define the required MBEDTLS_THREADING_C and  MBEDTLS_THREADING_PTHREAD : https://github.com/Homebrew/homebrew-core/blob/master/Formula/mbedtls.rb

**Comment by zaxebo1** [comment]
@pperidont : **how much time** it would take for you to implement or submit your pull request for the proposed patches/fixes ?
Actually, we are eagerly waiting for the new release for packaging with OSes. Last package(bundled in the Os repositories) is still 3 years old nekovm

**Comment by Simn** [comment]
Up?

**Comment by andyli** [comment]
Remaining tasks:
- [ ] https://github.com/HaxeFoundation/neko/pull/125
- [ ] bump version number (to 2.1?)
- [ ] prepare changelog
- [ ] actually make the release (upload binary and source archives, update website etc.)

@ncannasse Do you want me to handle those, or will you take care of them?

**Comment by ncannasse** [comment]
Please do, I don't have much time for it atm

**Comment by andyli** [comment]
Done. I will update the packages in #127.
